src/actions/exams.ts

```diff
@@ -5,28 +5,202 @@ import db from '@/lib/db'
 import { revalidatePath } from 'next/cache'
 import { cache } from 'react'
 
-export const getExams = cache(async () => {
-  const response = await db.exam.findMany({
-    select: {
-      id: true,
-      title: true,
-      description: true,
-      duration: true,
-      price: true,
-    },
-  })
-  return response
-})
+type CreateExamInput = {
+  title: string
+  description: string
+  duration: number
+  price: number
+  numQuestions: number
+}
+
+type UpdateExamInput = {
+  id: string
+  title?: string
+  description?: string
+  duration?: number
+  price?: number
+  numQuestions?: number
+}
 
 interface SubmitExamParams {
   examId: string
   answers: Record<string, number>
   timeSpent: number
   warningCount: number
+  questions: string[]
+}
+
+export const createExam = cache(async (examData: CreateExamInput) => {
+  try {
```

Comment:
Can u add authentication to all the actions

Reply:
Done